### PR TITLE
Us7 edit

### PR DESCRIPTION
**Pull request by rene-fas**

### CODE REVIEW

File: components/FieldTripListPage/index.js [hunk 1]
```diff
@@ -13,34 +13,44 @@ import {
...
-      // if input is not empty
-      const newDate = formatDate(newFieldTripDate); // format new date
+      const newDate = formatDate(newFieldTripDate);
 
       const newFieldTrip = {
-        id: fieldtrips.length + 1, // add new fieldtrip id
-        fieldtripname: newFieldTripName, // add new fieldtrip name
-        fieldtripdate: newDate, // add new fieldtrip date
-        outcrops: [], // add new fieldtrip outcrops
+        id: fieldtrips.length + 1, // added fieldtrip id is dependant on the length of fieldtrips
```
Comment (julialoeschel) [inline components/FieldTripListPage/index.js]: I sugest using uuid or any otehr id-generating library. 
imagine you are deleting one item later - its very likely that there are two same ids very fast.

File: components/FieldTripListPage/index.js [hunk 2]
```diff
@@ -116,6 +169,28 @@ const FieldTripListPage = () => {
...
         </Dialog>
       )}
+
+      {showEditPopup && (
+        <Dialog>
+          <h2>Edit Field Trip</h2>
+          <input
+            type="text"
+            value={editedFieldTripName}
+            onChange={(e) => setEditedFieldTripName(e.target.value)}
```
Comment (julialoeschel) [inline components/FieldTripListPage/index.js]:
```suggestion
            onChange={(event) => setEditedFieldTripName(event.target.value)}
```

File: components/FieldTripListPage/index.js [hunk 3]
```diff
@@ -116,6 +169,28 @@ const FieldTripListPage = () => {
...
+            placeholder="Field Trip Name"
+          />
+          <input
+            type="date"
+            value={editedFieldTripDate}
+            onChange={(e) => setEditedFieldTripDate(e.target.value)}
```
Comment (julialoeschel) [inline components/FieldTripListPage/index.js]: same here - we prefere to write everything out. better for future-you and better for any other develoeper in your team :)

File: components/OutcropDetailsPage/index.js [hunk 4]
```diff
@@ -94,8 +95,9 @@ const OutcropDetailsPage = () => {
   };
 
   const setStoredFieldTrips = (data) => {
+    //function to set stored field trips in local storage
     try {
-      localStorage.setItem("fieldTrips", JSON.stringify(data)); // set stored field trips
+      localStorage.setItem("fieldTrips", JSON.stringify(data));
```
Comment (julialoeschel) [inline components/OutcropDetailsPage/index.js:100]: for your local storage usage - I would suggest to use [use-local-storage-state](https://github.com/neuefische/muc-web-23-3/blob/main/sessions/react-with-local-storage/react-with-local-storage.md#use-local-storage-state) hook. check it out think about it (could be a later userstory)

File: components/OutcropListPage/index.js [hunk 5]
```diff
@@ -39,29 +44,31 @@ const OutcropListPage = ({ fieldtripId }) => {
...
 
         const newOutcrop = {
+          // Create new outcrop
           id: currentFieldTrip.outcrops.length + 1,
           name: newOutcropTitle,
           details: [],
         };
 
-        currentFieldTrip.outcrops.push(newOutcrop);
+        currentFieldTrip.outcrops.push(newOutcrop); // Add new outcrop to current field trip
```
Comment (julialoeschel) [inline components/OutcropListPage/index.js]: you can use the newer [spread-syntax](https://github.com/neuefische/muc-web-23-3/blob/main/sessions/js-modern-syntax/js-modern-syntax.md#spread-syntax-) here.

**Review by julialoeschel — COMMENTED** [review]
great work there! 
I made some suggestions to your code.

🐜

File: components/FieldTripListPage/index.js (hunk 1 above)
Comment (rene-fas) [reply components/FieldTripListPage/index.js]: Yeah, thanks kinda huge problem after testing it a bit solved it without a library or atleast i hope so.

**Review by MariaRiosNavarro — APPROVED** [review]

**Comment by MariaRiosNavarro** [comment]
I don't see any faults, everything was corrected as stated above. 🚀

**Pull request merged**